### PR TITLE
Zoma/testbot2

### LINKED ISSUE #4227: Set default values for topAnswer of QnA
## Version
What package version of the SDK are you using.

## Describe the bug
We need to pre-set startIndex and endIndex of topAnswer, or else it will show the following error:
image

## Expected behavior
Give a clear and concise description of what you expected to happen.

## Additional context

[bug]

### PR DESCRIPTION
**Pull request by coldplaying42**
Fixes #4227 

Add unit tests for QnAMakerExtensions.

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.AI.LUIS/LuisExtensions.cs [hunk 1]
```diff
@@ -29,7 +32,7 @@ public static IConfigurationBuilder UseLuisSettings(this IConfigurationBuilder b
             var luisRegion = configuration.GetValue<string>("LUIS_AUTHORING_REGION") ?? configuration.GetValue<string>("region") ?? "westus";
             var environment = configuration.GetValue<string>("environment") ?? Environment.UserName;
             var settings = new Dictionary<string, string>();
-            settings["luis:endpoint"] = $"https://{luisRegion}.api.cognitive.microsoft.com";
+            settings["luis:endpoint"] = configuration.GetValue<string>("luis:endpoint") ?? $"https://{luisRegion}.api.cognitive.microsoft.com";
```
Comment (gabog) [inline libraries/Microsoft.Bot.Builder.AI.LUIS/LuisExtensions.cs:35]: How is this related to the issue associated with this PR?

File: libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+using System.IO;
+using System.Text.RegularExpressions;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Bot.Builder.AI.QnA
+{
+    /// <summary>
+    /// Extension methods for QnA.
+    /// </summary>
+    public static class QnAMakerExtensions
```
Comment (gabog) [inline libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerExtensions.cs:16]: How is this related to the issue associated with this PR?

File: libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        /// <remarks>
+        /// This will pick up --environment as the environment to target.  If environment is Development it will use the name of the logged in user.
+        /// This will pick up --root as the root folder to run in.  
+        /// </remarks>
+        /// <param name="builder">Configuration builder to modify.</param>
+        /// <param name="botRoot">bot Root.</param>
+        /// <param name="qnaRegion">qnaRegion.</param>
+        /// <param name="environment">enviroment.</param>
+        /// <returns>Modified configuration builder.</returns>
+        public static IConfigurationBuilder UseQnAMakerSettings(this IConfigurationBuilder builder, string botRoot, string qnaRegion, string environment)
```
Comment (gabog) [inline libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerExtensions.cs:30]: I don't have a lot of context on this approach, but if this is trying to be similar to LuisExtensions, why doesn't it have the same arguments? shouldn't the look and work the same?

File: libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerRecognizer.cs [hunk 4]
```diff
@@ -209,9 +209,11 @@ public override async Task<RecognizerResult> RecognizeAsync(DialogContext dialog
                 ObjectPath.SetPathValue(recognizerResult, "entities.answer", answerArray);
 
                 var instance = new JArray();
-                instance.Add(JObject.FromObject(topAnswer));
+                var data = JObject.FromObject(topAnswer);
```
Comment (gabog) [inline libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerRecognizer.cs:212]: Need unit tests for this code.

File: libraries/Microsoft.Bot.Builder.AI.LUIS/LuisExtensions.cs (hunk 1 above)
Comment (coldplaying42) [reply libraries/Microsoft.Bot.Builder.AI.LUIS/LuisExtensions.cs:35]: I have discarded this change.

File: libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerExtensions.cs (hunk 3 above)
Comment (coldplaying42) [reply libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerExtensions.cs:30]: We do not want to revise the existing luis extension. But we hope to set the default values for qna maker extension  in testbot.json.

File: libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerRecognizer.cs (hunk 4 above)
Comment (coldplaying42) [reply libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerRecognizer.cs:212]: I have added the unit tests.

File: libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerExtensions.cs (hunk 2 above)
Comment (coldplaying42) [reply libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerExtensions.cs:16]: I have splitted this PR into two new PRs.
